05.wc/wc.rb

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = create_options
+  file_details = create_file_details
+
+  display_file_details(file_details, options)
+end
+
+def create_options
+  options = {}
+
+  OptionParser.new do |opts|
+    opts.on('-l') { |v| options[:lines] = v }
+    opts.on('-w') { |v| options[:words] = v }
+    opts.on('-c') { |v| options[:bytes] = v }
+  end.parse!
+
+  options.empty? ? { lines: true, words: true, bytes: true } : options
+end
+
+def create_file_details
+  file_name_and_contents = create_file_name_and_contents
+
+  file_details = file_name_and_contents.each.map { |file_name_and_content| create_file_details_hash(file_name_and_content) }
+  file_details << create_total_details(file_details) if file_details.size > 1
+
+  file_details
+end
+
+def create_file_name_and_contents
+  return [{ file_name: '', content: $stdin.read }] if ARGV.empty?
+
+  ARGV.map { |file_name| { file_name:, content: File.read(file_name) } }
+end
+
+def create_file_details_hash(file_content)
```

Comment:
file_detailsが「ハッシュの詰まった配列」なのであれば、ただのハッシュは file_detail と呼べばいいと思います。
すなわち、

```suggestion
def create_file_detail(file_content)
```

とした方が命名がシンプルにになります